src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/SubsetParameters.java

```diff
@@ -0,0 +1,47 @@
+package au.org.emii.gogoduck.worker;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import au.org.emii.gogoduck.worker.SubsetParameters.SubsetParameter;
+
+public class SubsetParameters extends HashMap<String, SubsetParameter> {
```

Comment:
No but you might be able to make use of http://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableMap(java.util.Map) or the like.